middleware/validation.js

```diff
@@ -0,0 +1,111 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const path = require('path')
+const fs = require('fs')
+
+/*
+  This method creates a joi vaidation schema.
+  It returns null if the schema creation fails and updates the context with an error
+*/
+const createJoiValidationSchema = (ctx) => {
```

Comment:
Can we remove 'Joi' from the name? 
That allows us to swap out the validation module/framework for something else and not have the name referring to a specific library.